src/pages/actions/useSideEffect.tsx

```diff
@@ -15,16 +16,26 @@ export default function useSideEffect(
 
     switch (sideEffect.type) {
       case "reloadData": {
+        setPreventReload(false);
         reload();
         return;
       }
       case "executeAction": {
+        setPreventReload(true);
         executeAction(sideEffect.payload.action, sideEffect.payload.data).then(
           response => dispatch(actionSuccessful(response)),
           error => dispatch(actionFailed(error))
         );
         return;
       }
+      case "allowReload": {
```

Comment:
:mega: As mentioned in person, I think this should be another `useEffect` next to this one that depends on the two dialog variables changing.